### PR TITLE
Remove ubuntu-boot and replace with a simpler mbr

### PR DESCRIPTION
**Pull request by valentindavid**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -55,15 +55,7 @@ GRUB_MODULES = \
 	true \
 	video
 
-all:
```
Comment (xnox) [inline Makefile:58]: everything above can be removed as well given this change.

**Review by xnox — APPROVED** [review]
In the Makefile, lists of grub modules to vendor no longer needed and should be removed. Only static grub.efi binary is needed now.

In snapcraft.yaml you can then also drop "build-packages"   [ grub-pc-bin, grub-common ]

**Review by xnox — CHANGES_REQUESTED** [review]
See above requests to remove even more things given this change.

File: Makefile (hunk 1 above)
Comment (valentindavid) [reply Makefile:58]: Done.

File: legacy-bios/mbr.s [hunk 2]
```diff
@@ -0,0 +1,28 @@
+        .code16
+        .section .text
+        .globl start
+
+start:
```
Comment (Meulengracht) [inline legacy-bios/mbr.s:5]: Usually it's good practice to clear the code-segment as the first thing in bootloaders, you can't trust that the code-segment is valid. Adding the following lines is enough (sorry for my intel syntax):

```
start:
	jmp 0x0:entry

entry:
```

File: legacy-bios/mbr.s [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        xorw %ax, %ax
+        movw %ax, %ss
+        movw %ax, %ds
+        movw $0x7c00, %sp
```
Comment (Meulengracht) [inline legacy-bios/mbr.s:12]: While setting up segments it's a good idea to do this with interrupts disabled (don't want to modify SS without SP and then get an interrupt in-between)

```suggestion
        cli
        xorw %ax, %ax
        movw %ax, %ss
        movw %ax, %ds
        movw $0x7c00, %sp
        sti
```

File: legacy-bios/mbr.s (hunk 2 above)
Comment (valentindavid) [reply legacy-bios/mbr.s:5]: Thank you. That was `jmp $0, $entry`.

File: legacy-bios/mbr.s (hunk 3 above)
Comment (valentindavid) [reply legacy-bios/mbr.s:12]: Done.

File: legacy-bios/mbr.s [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+start:
+        jmp $0, $.entry
+.entry:
+        cli
+        xorw %ax, %ax
+        movw %ax, %ss
+        movw %ax, %ds
+        movw $0x7c00, %sp
+        sti
```
Comment (Meulengracht) [inline legacy-bios/mbr.s:13]: Last comment, and then it's a +1 from me, I missed this on initial review, but add a 'cld' instruction before printing the string. The 'cld' instruction clears the direction flag which is a good thing to do initially as we don't know the state of the direction flag when loaded. This affects how  `lodsb` loads string data.

```suggestion
        sti
        cld
```

**Review by mvo5 — APPROVED** [review]
Nice, thank you!

**Review by xnox — APPROVED** [review]

File: legacy-bios/mbr.s (hunk 4 above)
Comment (valentindavid) [reply legacy-bios/mbr.s:13]: Done!

**Review by Meulengracht — APPROVED** [review]
Thanks for the changes, LGTM!

**Review by alfonsosanchezbeato — APPROVED** [review]
Very nice, LGTM

**Pull request merged**